### PR TITLE
Added boosts to LorentzVectorImpl

### PR DESCRIPTION
**Pull request by apBUSampK**

### CODE REVIEW

File: LorentzVector.hh [hunk 1]
```diff
@@ -57,6 +57,38 @@ namespace cola {
             return *this;
         }
 
+        // borrowed from CLHEP. bx by and bz are projections of beta
+        LorentzVectorImpl& boost(Type bx, Type by, Type bz) {
+            Type b2 = bx*bx + by*by + bz*bz;
+            if (b2 >= 1)
+                throw std::runtime_error("Boost faster than speed of light.");
+            Type ggamma = 1.0 / std::sqrt(1.0 - b2);
```
Comment (alexsvetlichnyy) [inline LorentzVector.hh]: will lead to a significant error for HEP calculations as beta2 approx 1

File: LorentzVector.hh [hunk 2]
```diff
@@ -57,6 +57,38 @@ namespace cola {
...
+
+        // axis from 1 to 3 correspond to x-y-z
+        LorentzVectorImpl& boostAxis(Type beta, uint axis=3) {
+            Type b2 = beta*beta;
+            if (axis > 3 or axis < 1) {
+                throw std::runtime_error("Wrong axis in boostAxis. 1 for x, 2 for y, 3 for z.");
+            }
+            if (b2 >= 1)
+                throw std::runtime_error("Boost faster than the speed of light.");
+            Type ggamma = std::sqrt(1./(1.-b2));
```
Comment (alexsvetlichnyy) [inline LorentzVector.hh]: Same as above

File: LorentzVector.hh [hunk 3]
```diff
@@ -57,6 +57,38 @@ namespace cola {
...
         }
 
+        // borrowed from CLHEP. bx by and bz are projections of beta
+        LorentzVectorImpl& boost(Type bx, Type by, Type bz) {
+            Type b2 = bx*bx + by*by + bz*bz;
+            if (b2 >= 1)
+                throw std::runtime_error("Boost faster than speed of light.");
+            Type ggamma = 1.0 / std::sqrt(1.0 - b2);
+            Type bp = bx*x + by*y + bz*z;
+            Type gamma2 = b2 > 0 ? (ggamma - 1.0)/b2 : 0.0;
```
Comment (alexsvetlichnyy) [inline LorentzVector.hh]: what is gamma2?

Comment (apBUSampK) [reply LorentzVector.hh]: A common factor in matrix of general Lorentz transformation

File: LorentzVector.hh (hunk 1 above)
Comment (apBUSampK) [reply LorentzVector.hh]: Honestly, do we have another choice without using rapidity?

Comment (alexsvetlichnyy) [reply LorentzVector.hh]: nope, so use rapidity)

File: LorentzVector.hh (hunk 3 above)
Comment (alexsvetlichnyy) [reply LorentzVector.hh]: Okay

File: LorentzVector.hh (hunk 1 above)
Comment (apBUSampK) [reply LorentzVector.hh]: Ok...)

**Comment by apBUSampK** [comment]
Time's due, @alexsvetlichnyy will ensure validity later

**Pull request merged**